plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/aps/system/services/mapping/SeoMappingManager.java

```diff
@@ -71,19 +72,21 @@ public void init() throws Exception {
 	@Override
 	public void updateFromPageChanged(PageChangedEvent event) {
 		IPage page = event.getPage();
+        if (!PageChangedEvent.EVENT_TYPE_SET_PAGE_OFFLINE.equals(event.getEventType()) && 
+                !PageChangedEvent.EVENT_TYPE_SET_PAGE_ONLINE.equals(event.getEventType())) {
+            return;
+        }
 		try {
 			if (event.getSource().equals(ApsEvent.LOCAL_EVENT)) {
-				if (null == this.getPageManager().getOnlinePage(page.getCode())) {
-					this.deleteMapping(page.getCode(), null);
-				} else {
+                this.deleteMapping(page.getCode(), null);
```

Comment:
Is this logical change intentional? Seems like an exception waiting to happen if the page isn't online? As long as it is covered by tests it is ok but want to check

Reply:
I've changed also this function